crates/tauri/src/app.rs

```diff
@@ -685,6 +685,32 @@ macro_rules! shared_app_impl {
         })
       }
 
+      /// Set the app theme.
+      pub fn set_theme(&self, theme: Option<Theme>) {
+        #[cfg(windows)]
+        for window in self.manager.windows().values() {
+          if let (Some(window_menu), Ok(hwnd)) = (&*window.menu_lock(), window.hwnd()) {
+            let raw_hwnd = hwnd.0 as isize;
+            let menu = window_menu.menu.clone();
```

Comment:
instead of holding the `menu_lock` here for the rest of the scope, we can just clone and release the lock
```suggestion
          let window_menu = window.menu_lock();
          if let (Some(menu), Ok(hwnd)) = (window_menu, window.hwnd()) {
            let raw_hwnd = hwnd.0 as isize;
```